### ISSUE TITLE
Not able to select input channel 2 in Focusrite Scarlett solo 3rd gen

### ISSUE THREAD

**Issue by rakeshcoresoft** [sdatkinson/NeuralAmpModelerPlugin#157]
I have Focusrite Scarlett Solo 3rd Gen as my  audio interface. It uses Input channel 2 as Instrument input. It is not possible to select input channel 2 in NAM Standalone GUI. Please add this feature.

**Comment by danielbeutner** [comment]
I do have the same problem with the arturia minifuse 2. Only channel 1 works for NAM standalone and channel 2's volume seems too low.

**Comment by seve54** [comment]
I do have the same problem with audio ìnterface Terratec Phase 22

**Comment by DungeonDev78** [comment]
Same problem here :( Input device is just Scarlet Solo USB without input 1 or 2.

**Comment by karymus** [comment]
Same with M AUDUO SOLO.

**Comment by sdatkinson** [comment]
This should be fixed with #166 and will be available in v0.7.3.

**Comment by sdatkinson** [comment]
Bad news I'm afraid: the fix breaks support for folks with single-input interfaces.

It's a tough call, but I've reverted the fix for this. I'll have to ask folks to continue to sit tight on this one while we work on getting the right fix in.

**Comment by nmontua** [comment]
Would it be too messy to have to builds for the time being?

**Comment by sdatkinson** [comment]
> Would it be too messy to have two builds for the time being?

I've _strongly_ considered it.

The reason that tipped me to the "no" side is that it'd open me up to potentially a whole other source of bugs to have to field, and it might be especially hard to get folks to provide the needed information (which version they're using). I just don't think I could handle it.

**Comment by nmontua** [comment]
> > Would it be too messy to have two builds for the time being?
> 
> I've _strongly_ considered it.
> 
> The reason that tipped me to the "no" side is that it'd open me up to potentially a whole other source of bugs to have to field, and it might be especially hard to get folks to provide the needed information (which version they're using). I just don't think I could handle it.

Fair enough, I just thought I'd ask.

**Comment by RocklandSally** [comment]
I should have come here first .. lol.. yep.. same problem with ONLY the standalone version of NAM... within Reaper you can specify input #2 ... thanks for everything @sdatkinson

**Comment by twynfeyr** [comment]
I can see you've identified the problem and a solution is incoming, but I'll add my interface to the list in case it matters in some way.  Steinburg UR12.

**Comment by olilarkin** [comment]
this is an iPlug2 bug, please add the iplug2 label!

**Comment by jonlinnarson** [comment]
> this is an iPlug2 bug, please add the iplug2 label!

Does that mean that Steven can't really do anything about this issue?

**Comment by sdatkinson** [comment]
@jonlinnarson kind of.

Technically, I could fix the problem in iPlug2. On one hand, I'm not an expert in the framework; on the other hand, I can reproduce this Issue myself, which makes a huge difference in how hard it is to figure out what to do about it! 🙂

**Comment by splisp** [comment]
Yes, that's really unfortunate. I have a Yamaha AG03 which has an hi-Z instrument input on channel 2 and it cannot be selected. I have also entered another issue on iplug2's preferences dialog a few minutes ago.

**Comment by olilarkin** [comment]
FYI: I'm working on fixing the standalone app I/O here: https://github.com/iPlug2/iPlug2/tree/app/flexible-io

**Comment by andylockran** [comment]
@olilarkin if you have a PR you're mid-through, can you open so I can fork and help write tests/debugs .etc?

**Comment by enricostragiotti** [comment]
Any news on that problem?

**Comment by seve54** [comment]
No
________________________________

**Comment by Gimmee99** [comment]
+1 on getting a fix from me.  I have the Tascam 16x8 interface and port 1 in an XLR input.  1/4" inputs don't start until 9.

**Comment by fatgonzo** [comment]
Having the same issue with Steinberg UR22, ideally would like to use input 2 for both channels.

**Comment by Gimmee99** [comment]
@sdatkinson Do you know if there is a specific bug for this in the ipug project?  I see above where someone says they are "working on this" with a link, but I think for me that just links to the general project and not a specific issue.  I have *some* experience with software engineering and might be able to either help with this or find some people to help me with it if nobody else is going to ever get this fixed?
